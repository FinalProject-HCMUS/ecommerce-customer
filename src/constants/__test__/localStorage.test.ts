import localStorageConstants from '../localStorage';

describe('localStorage constants', () => {
  test('should export an object with all expected keys', () => {
    expect(localStorageConstants).toBeDefined();
    expect(typeof localStorageConstants).toBe('object');
    const expectedKeys = [
      'LOCALIZATION',
      'TOKEN',
      'REFRESH_TOKEN',
      'EMAIL',
      'USER_INFO',
    ];
    expectedKeys.forEach((key) => {
      expect(localStorageConstants).toHaveProperty(key);
    });
  });

  test('should have correct values for each key', () => {
    expect(localStorageConstants.LOCALIZATION).toBe('localization');
    expect(localStorageConstants.TOKEN).toBe('token');
    expect(localStorageConstants.REFRESH_TOKEN).toBe('refresh_token');
    expect(localStorageConstants.EMAIL).toBe('email');
    expect(localStorageConstants.USER_INFO).toBe('user_info');
  });

  test('should match snapshot', () => {
    expect(localStorageConstants).toMatchSnapshot();
  });
});
