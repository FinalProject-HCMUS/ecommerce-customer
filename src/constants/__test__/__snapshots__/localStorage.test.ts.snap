// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`localStorage constants should match snapshot 1`] = `
{
  "EMAIL": "email",
  "LOCALIZATION": "localization",
  "REFRESH_TOKEN": "refresh_token",
  "TOKEN": "token",
  "USER_INFO": "user_info",
}
`;
